service/controller/resource/controlplanestatus/resource.go

```diff
@@ -70,7 +70,7 @@ func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
 		}
 	}
 	masterNodes, err := r.nodeCount.MasterCount(ctx, cr)
-	if nodecount.IsTenantClusterInitialized(err) {
+	if nodecount.IsNotTenantClusterInitialized(err) {
```

Comment:
For consistency's sake, when the error is called `tenantClusterNotInitializedError`, the matcher naming should be aligned to that. In this case the meaning gets also slightly twisted. So `IsTenantClusterNotInitialized` would be the correct name. 

Another thing that is slightly off is that there is also this. 

```
	tenantClient, err := r.tenantClient.K8sClient(ctx, cr)
	if tenantclient.IsNotAvailable(err) {
```

So there is this error but then the node count service does it differently. It looks to me like we could make it simpler and be transparent about the tenant client error. Not a deal breaker here but maybe food for thought.

Reply:
I changed the naming of IsTenantClusterNotInitialized, my bad didn't saw it in the first place thanks.

Regarding the error handling I agree, I changed it to return the error directly instead of changing it again. 

I added in `controlplanestatus` and `machinedeploymentstatus` resource that it checks both errors in case it happens:
`if tenantclient.IsNotAvailable(err) || nodecount.IsTenantClusterNotInitialized(err)`

I hope this is fine for you.